File: test-suites/acceptance/testkit/model/customer.py
from mimesis import Person, Address as FakeAddress
from mimesis.random import Random


class Address:
    street = ""
    city = ""
    state = ""
    zip = ""

    def __init__(self, **kwargs):
        self.__dict__.update(kwargs)

    @staticmethod
    def fake():
        addr = Address()
        fake_addr = FakeAddress()

        addr.__dict__.update({
            "street": fake_addr.street_name(),
            "city": fake_addr.city(),
            "state": fake_addr.state(),
            "zip": fake_addr.zip_code(),
        })
        return addr


class Customer:
    id = ""
    login = ""
    password = ""
    first_name = ""
    last_name = ""
    phone = ""
    ssn = ""
    address = Address()

    def __init__(self, **kwargs):
        self.__dict__.update(kwargs)

    @staticmethod
    def fake():
        fake_person = Person()
        customer = Customer()

        customer.__dict__.update({
            "id": Random().custom_code(),
            "login": fake_person.username(mask='l'),
            "password": fake_person.username(mask='l'),
            "first_name": fake_person.first_name(),
            "last_name": fake_person.last_name(),
            "phone": fake_person.telephone(),
            "ssn": Random().custom_code(),
            "address": Address.fake(),
        })
        return customer

File: test-suites/e2e/step_definitions/testkit/models/customer.ts
import {faker} from '@faker-js/faker';

export class Customer {
    public username: string;
    public password: string;

    public firstName: string;
    public lastName: string;
    public street: string;
    public city: string;
    public state: string;
    public zip: string;
    public phone: string;
    public ssn: string;

    /**
     * Returns a model prefilled with fake data.
     */
    public static getFake() {
        const c = new Customer();
        c.username = faker.internet.userName();
        c.password = faker.random.alphaNumeric(10);
        c.firstName = faker.name.firstName();
        c.lastName = faker.name.lastName();
        c.street = faker.address.streetAddress();
        c.city = faker.address.city();
        c.state = faker.address.state();
        c.zip = faker.address.zipCode();
        c.phone = faker.phone.number();
        c.ssn = faker.phone.imei();
        return c;
    }
}